rasa/core/lock_store.py

```diff
@@ -0,0 +1,227 @@
+import asyncio
+import json
+import logging
+import typing
+from typing import Text, Optional, Dict, Union
+
+from async_generator import asynccontextmanager, async_generator, yield_
+
+from rasa.core.lock import TicketLock
+
+if typing.TYPE_CHECKING:
+    pass
+
+logger = logging.getLogger(__name__)
+
+ACCEPTED_LOCK_STORES = ["in_memory", "redis"]
+
+
+class LockError(Exception):
+    pass
+
+
+class LockStore(object):
+    def __init__(self, lifetime: int = 60) -> None:
```

Comment:
I think we never have these for constructors
```suggestion
    def __init__(self, lifetime: int = 60):
```